sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientRegistration.cs

```diff
@@ -3,11 +3,12 @@
 
 using System;
 using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
 using Azure.Core;
 
 namespace Microsoft.Extensions.Azure
 {
-    internal class ClientRegistration<TClient>
+    internal class ClientRegistration<TClient> : IDisposable, IAsyncDisposable
```

Comment:
I'd really like to add a `class` constraint to the generic here to enforce a reference type, but I believe that would be a breaking change.    I'm afraid that the factory already doesn't work with value types, however, as L48 would never evaluate to `true`...  am I overlooking something?

Reply:
Confirming, this does not work with a `struct` or `int` due to the factory assuming `null` for no value.

Reply:
To me that seems like a justifiable breaking change since presumably no customers would be able to pass anything in besides a class - so this would already be indirectly enforced. Unless there's some kind of weird case with a different ref type that this could break? To me this seems like an improvement to usability since it will make it faster for customers to see errors.

Reply:
I think I'm going to treat that as a potential second effort and open discussion with the architects.   It doesn't directly impact the support for disposable, which was the main goal here.